### PR TITLE
add persian language translate to luya framework

### PR DESCRIPTION
**Pull request by masihfathi**
i add Persian language translate to luya framework. but one thing remain, the Persian language is right to left direction. is it possible to add rtl direction to editor and template for rtl languages?

| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    |no
| Tests pass?   | not test

### CODE REVIEW

**Comment by nadar** [comment]
hi @masihfathi, well first of all: thank you! I saw some translations are still missing, should i keep the PR open?

To be honest i am not familiar with RTL writiting. Can you make some suggestion how we should do this in textareas and inputs (or any other elements?)

File: envs/kickstarter/messages/fa/app.php [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?php
+return [
```
Comment (nadar) [inline envs/kickstarter/messages/fa/app.php]: Missing translation

File: modules/admin/src/messages/fa/admin.php [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    'model_group_description' => 'توضیحات',
+    'model_group_user_buttons' => 'کاربر',
+    'model_group_btn_aws_groupauth' => 'سطح دسترسی ها',
+    'ngrest_crud_btn_list' => 'ورودی ها',
+    'ngrest_crud_btn_add' => 'اضافه',
+    'ngrest_crud_btn_close' => 'بستن',
+    'ngrest_crud_search_text' => 'عبارت مورد نظر خود را برای جستجو وارد کنید',
+    'ngrest_crud_rows_count' => 'ورودی ها',
+    'ngrest_crud_btn_create' => 'ایجاد',
+    'api_storage_image_upload_error' => 'The following error occurred while uploading an image \'{error}\'.',
```
Comment (nadar) [inline modules/admin/src/messages/fa/admin.php]: Missing translation ... some more below

File: modules/cms/src/admin/messages/fa/cmsadmin.php [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    'view_index_add_type' => 'نوع صفحه',
+    'view_index_type_page' => 'صفحه',
+    'view_index_type_module' => 'ماژول',
+    'view_index_type_redirect' => 'فروارد',
+    'view_index_as_draft' => 'به عنوان قالب',
+    'view_index_as_draft_help' => 'آیا می خواهید صفحه جدیدی به عنوان قالب تعریف کنید؟',
+    'view_index_no' => 'خیر',
+    'view_index_yes' => 'بله',
+    'view_index_page_title' => 'عنوان صفحه',
+    'view_index_page_alias' => 'URL Path segment',
```
Comment (nadar) [inline modules/cms/src/admin/messages/fa/cmsadmin.php]: Missing translation, some more below

**Comment by masihfathi** [comment]
Some words do not have equivalent meanings in Persian language and they should be in English.

**Comment by masihfathi** [comment]
We should consider some thing in RTL languages like Persian and Arabic.
See this article:
https://www.upwork.com/hiring/for-clients/web-design-for-rtl-languages-basics/

**Comment by nadar** [comment]
Hmm okay: but do you have any concret input what we have to change now?

**Comment by masihfathi** [comment]
for now and in this release and cause of short time to first release, i think at least it should be good to add rtl and ltr button to the text editor element like ckeditor or tinymce buttons and other complex thing like rtl bootstrap i think it can not be done in short time.

**Comment by nadar** [comment]
We are not using any WYSIWYG editor, and i don't think we are offering a RTL or LTR button, i thinking more about a config option for the user. So this is all we have to set on textarea and text's: https://www.w3schools.com/tags/att_global_dir.asp ? Whats your experience with this?

**Comment by masihfathi** [comment]
for now it is enough. and about justify element do you have any plan?

**Comment by nadar** [comment]
What do you mean exactly with "justify element"?

**Comment by masihfathi** [comment]
i mean sth like this in the text editor of the luya.

**Comment by nadar** [comment]
@masihfathi We are not like any other cms system, we don't believe in WYSIWG editors, we don't want messy output, we don't want that customers can write in red when the CI Design is in blue, WYSIWG are a pain. a website is NOT like a WORD document - you, as a developer, should know this :-)

Use blocks, structured elements, you can do anything with them. Move, copy, move between to other paragraphs, add structured data to given input, mix with database data.

If you still are not happy you can add your own block with tinyMCE ;-)

**Comment by masihfathi** [comment]
you are right, i know that, i mean simple element like Align and direction not all element of tinyMCE or ... .

**Comment by nadar** [comment]
I can not see any difference between those images.

I am going to merge this pull request, thank you again, its great! Lets discuss about LTR, RTL in a separate issue.

**Pull request merged**